### DISCUSSION TITLE
Typography

### DISCUSSION THREAD

**Discussion by bameyrick** [ONSdigital/design-system#233] [Feedback]
Use this issue to discuss [typography](https://ons-design-system.netlify.com/styles/typography) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by bameyrick** [comment]
Currently the default font sizes for headings are always one step larger than we actually use by default, meaning everywhere we use a heading we have to use a font size utility class. We should change the default font sizes for headings to what we set them to with utility classes, and then remove utility classes from the headings.

**Reply by jrbarnes9** [reply]
Resolved in #371

**Comment by jrbarnes9** [comment]
H4 headings are the same at the base font style, so it's impossible to differentiate between a paragraph and an h4 heading.

**Reply by jrbarnes9** [reply]
Resolved #379

**Comment by bameyrick** [comment]
The spacing between numbers are off. The font files need to be regenerated with "Proportional Figures" turned on.

Proportional Figures off:
![Screenshot 2019-06-06 at 15 34 48](https://user-images.githubusercontent.com/8447446/59041538-b40d5c80-8870-11e9-8b1f-11429925308a.png)

Proportional Figures on:
![Screenshot 2019-06-06 at 15 34 56](https://user-images.githubusercontent.com/8447446/59041540-b40d5c80-8870-11e9-9fac-039857b9cad8.png)

**Reply by jrbarnes9** [reply]
Resolved #409

**Comment by jeremygillies** [comment]
The .ttf format of Open Sans can implement `font-variant-numeric: tabular-nums;` which activates the set of figures where numbers are all of the same size, allowing them to be easily aligned in tables. It's required according to the ONS Styleguide : https://style.ons.gov.uk/category/data-visualisation/tables/#fonts

Our version doesn't implement it.

OpenSans (.woff) 
![image](https://user-images.githubusercontent.com/1080383/66906436-9e1e5e00-efff-11e9-8a09-ff8d7a4507ec.png)

Open Sans (.ttf)
![image](https://user-images.githubusercontent.com/1080383/66906477-b2faf180-efff-11e9-84f5-62289c173eee.png)

Looks like our font doesn't contain the monospaced `1` glyph

See what glyphs are missing by dropping our font in here ... https://fontdrop.info/

**Reply by jrbarnes9** [reply]
Resolved #640

**Comment by jrbarnes9** [comment]
Add external link guidance including warning users (with text) when an link will open in a new window (web chat example)

**Reply by jrbarnes9** [reply]
Issue #773

**Comment by jrbarnes9** [comment]
Define how postal address should be marked up/displayed

**Reply by jrbarnes9** [reply]
Issue #800 resolved

**Comment by jrbarnes9** [comment]
We shouldn't use colour as the only indicator of hover states on links, accordions and buttons etc 

https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html

**Reply by jrbarnes9** [reply]
Issue #803 resolved

**Comment by trevorsaint** [comment]
I’ve spotted a bug with the bare list that needs fixing. 

If we need the full width to be clickable, there are tricks to make this happen without the full width focus style. This may have been done for mobile benefits to allow a greater touch area.

![image](https://user-images.githubusercontent.com/4989027/87309891-1d9c7780-c515-11ea-935b-8fcb13d7fd7b.png)

**Reply by jrbarnes9** [reply]
Issue #833 resolved

**Comment by trevorsaint** [comment]
### Typography is rendering differently in Firefox than it is in Chrome

**We have** 
`-webkit-font-smoothing: antialiased;` 

**Missing** 
`-moz-osx-font-smoothing: grayscale;`

Adding this should fix the issue, ensuring typography consistency.

**Chrome**
![image](https://user-images.githubusercontent.com/4989027/88310178-42f46700-cd07-11ea-844f-4c3b3105e30b.png)

**Firefox rendering thicker**
![image](https://user-images.githubusercontent.com/4989027/88310142-37a13b80-cd07-11ea-8fbd-108eb608cfb2.png)

**Reply by jrbarnes9** [reply]
Issue #851 resolved

**Comment by Paul-Joel** [comment]
The prefix option in lists no longer works after #892 (released in v22.0.0)
The issue seems to be that 'prefixText' has been introduced to replace 'prefix', but 'prefix' still exists, and the code checks for 'prefix', then writes out 'prefixText'
The prefix setting is in the Census help pages, and is now broken.
It should also have been marked as a breaking change, as Craft was using 'prefix', which no longer displays the value.
There is an earlier possible issue as well that if a prefix is specified, you additionally have to manually declare 'list--bare' to stop it writing out both the automatic bullets and the prefix - this used to be added automatically

**Reply by jrbarnes9** [reply]
Issue #1011 resolved

**Comment by jrbarnes9** [comment]
Dashed list example missing from list styles

**Reply by jrbarnes9** [reply]
Resolved in #1022

**Comment by trevorsaint** [comment]
A property added to `.list__item` is causing list item content to not break when expected.

`white-space: nowrap;` is going to prevent a break in typography. This may have been added to fix other functionality but can’t be added at the global level.

**Current behaviour**
![image](https://user-images.githubusercontent.com/4989027/94565253-e1013200-0260-11eb-84e7-5c4020fec44c.png)

**Expected behaviour**
![image](https://user-images.githubusercontent.com/4989027/94565298-f1b1a800-0260-11eb-9a85-e612fcc04e9f.png)

**Reply by jrbarnes9** [reply]
Issue #1048 resolved

**Comment by trevorsaint** [comment]
When applying icon modifiers, for example: `svg-icon--m, l, xl` and so on we seem to be adding a margin top. 

When used in lists with icons we have a problem with alignment due to a margin top being applied.

**A standard list with icons looks like this:**

<img width="295" alt="Screenshot 2020-10-05 at 10 29 30" src="https://user-images.githubusercontent.com/4989027/95063025-b3e7d000-06f5-11eb-99a1-a6bf6545c790.png">

Applying a size modifier you get this:

<img width="303" alt="Screenshot 2020-10-05 at 10 31 45" src="https://user-images.githubusercontent.com/4989027/95063223-fdd0b600-06f5-11eb-88f5-197cafa31c3b.png">

I think to keep flexibility with icon scale we should remove the margin attribute.

### This has a knock on effect with using icons in panels
Alignment issues exist which is why margin top was added initially.

It may be a better idea to add utility classes for finer alignment tweaks, rather than adding it directly to the icon scale modifier.

**Reply by jrbarnes9** [reply]
Issue #1063 resolved

**Comment by trevorsaint** [comment]
I need the ability to define `rel` attributes for list links.

For example, social links may want something along the lines of `rel="noreferrer external"`.
<img width="384" alt="Screenshot 2020-10-13 at 10 48 06" src="https://user-images.githubusercontent.com/4989027/95845045-99d76e80-0d41-11eb-9eb6-9ffdd17dac79.png">

**Reply by jrbarnes9** [reply]
Issue #1087 resolved

**Comment by trevorsaint** [comment]
List items not clearing correctly.

Rather than using floats for the icons we should consider using flexbox. If this is not possible due to browser compatibility limitations, we will need to add a clear fix for the list items.

**See the issue below**
<img width="536" alt="Screenshot 2020-10-14 at 11 52 13" src="https://user-images.githubusercontent.com/4989027/95979753-0884fc80-0e14-11eb-97be-8d08c153f2cb.png">

**Reply by jrbarnes9** [reply]
Issue #1094 resolved

**Comment by trevorsaint** [comment]
Inline lists don’t play nicely with icons.

<img width="601" alt="Screenshot 2020-10-19 at 09 43 00" src="https://user-images.githubusercontent.com/4989027/96422167-9a637f80-11ef-11eb-9e9c-0d9a8164ed77.png">

We seem to be specifying `.svg-icon` size specifically for use within inline lists. We don’t need this. It is going to override icon scale as you can see in the example. applying a modifier in the case won’t have an effect due to specificity.

<img width="346" alt="Screenshot 2020-10-19 at 09 44 27" src="https://user-images.githubusercontent.com/4989027/96422321-c67f0080-11ef-11eb-9105-813d1b9da3fe.png">

**Removing the attribute** works just fine as the example below shows. The gap however is far too big between inline list items.
<img width="592" alt="Screenshot 2020-10-19 at 09 44 42" src="https://user-images.githubusercontent.com/4989027/96422667-38574a00-11f0-11eb-895b-2538a061a89e.png">

**Reply by jrbarnes9** [reply]
Issue #1103 resolved

**Comment by jrbarnes9** [comment]
Need a way to adjust quote component to reduce font size/style to allow for longer block quotes e.g. a quote from a director within a press release

**Reply by jrbarnes9** [reply]
Issue #1141 resolved

**Comment by jrbarnes9** [comment]
Add the ability to also set suffixIcon as well as `prefixIcon`

**Reply by jrbarnes9** [reply]
Issue #1173 resolved

**Comment by jrbarnes9** [comment]
I think we should consider revising the hover styling for inline body links.  We currently remove the underline (to meet WCAG 'don't rely on colour alone' guidance) but, some users have reported the links 'disappearing' during Census UR.  This is due to both the underline being removed, and the link hover colour being black in census theme.

Proposing to keep the underline on hover and increase thickness to 2px using `text-decoration-thickness` as an enhancement

https://user-images.githubusercontent.com/43346934/102881064-36a24080-4444-11eb-85ed-cb37279afad5.mov

**Reply by armstrongb** [reply]
Would moving from 1px to 2px satisfy someone who was having difficulty in discerning the colour contrast difference?

**Reply by jrbarnes9** [reply]
This new hover styling was part of the DAC audit in Jan 2021, and the thickness of the hover state underline was not raised as an issue, but I have put this on our backlog for focussed testing.

**Reply by jrbarnes9** [reply]
Issue #1261 resolved

**Comment by jrbarnes9** [comment]
Typography design forum link still linking to old issue

**Reply by jrbarnes9** [reply]
Issue #1262 resolved

**Comment by jrbarnes9** [comment]
`list--icons` negative margin causing overflow issue

Double-scroll bar present on census.gov.uk/contact-us in Edge 82 and Chrome 86.0.4240.111

![2 scroll bars](https://user-images.githubusercontent.com/43346934/106260983-f4290c00-6218-11eb-8763-f1a910030998.png)

**Reply by jrbarnes9** [reply]
Issue #1341 resolved

**Comment by jrbarnes9** [comment]
External links don't wrap

![Screenshot 2021-05-20 at 16 12 24](https://user-images.githubusercontent.com/43346934/119004208-42a81500-b986-11eb-9f9b-61746a8b9949.png)

**Reply by jrbarnes9** [reply]
Issue raised: #1530

**Reply by jrbarnes9** [reply]
Issue #1524 resolved

**Comment by jrbarnes9** [comment]
Enable the list macro to call the external link macro 

![Screenshot 2021-06-30 at 16 19 22](https://user-images.githubusercontent.com/43346934/123987351-045b3800-d9bf-11eb-98fc-a3dbb717f217.png)

**Reply by jrbarnes9** [reply]
Issue #1579 raised

**Comment by jrbarnes9** [comment]
Lists: `item.listclass` is defined in the macro but `classList` is specified in the macro options table.

**Reply by jrbarnes9** [reply]
Issue #1635 raised

**Comment by jrbarnes9** [comment]
Enable list component to cater for nested lists

**Reply by jrbarnes9** [reply]
Issue #1647 raised

**Comment by jrbarnes9** [comment]
We should conform icon usage so they're always set in the same way e.g.
- `params.iconType`
- `params.iconSize`
- `params.iconPosition`

Currently, buttons, panels and lists all set icons in a slightly different way.

**Reply by jrbarnes9** [reply]
Issue #1653 raised

**Comment by jrbarnes9** [comment]
All indented text/elements should follow a consistent style of border

Includes:
- `onsTextIndent`
- `onsCollapsible`
- Borderless `onsRadios` and `onsCheckboxes` with conditionally revealed input, select, checkboxes, radios

For the border colour, should we use the variable assigned to`$color-borders` or assign a lighter shade as a new variable? 

Current `onsTextIndent`:

<img width="321" alt="Screenshot 2021-08-24 at 16 39 04" src="https://user-images.githubusercontent.com/43346934/130648758-1469d62b-09f1-4d3d-8550-d4465ac98fb1.png">

Current `onsCollapsible`:

<img width="338" alt="Screenshot 2021-08-24 at 16 39 08" src="https://user-images.githubusercontent.com/43346934/130648756-c4df1293-5400-4b6f-af76-608b57516b2b.png">

Example of borderless `onsRadios` with conditionally revealed input:

![Screenshot 2021-08-24 at 16 44 27](https://user-images.githubusercontent.com/43346934/130648752-5c9b8996-012d-48cd-979e-772d39e2cc3a.png)

For an example of borderless `onsCheckboxes` with conditionally revealed checkboxes see #1648

**Reply by jrbarnes9** [reply]
Issue #1657 raised

**Comment by jrbarnes9** [comment]
Quote component missing macro options table

**Reply by jrbarnes9** [reply]
Issue #1675 raised

**Comment by jrbarnes9** [comment]
Single list items with urls are not rendering links

![Screenshot 2022-02-03 at 11 06 01](https://user-images.githubusercontent.com/43346934/152331432-8e1abee3-3ec7-423a-b030-a8dc787fde78.png)

**Reply by jrbarnes9** [reply]
Raised #1911

**Comment by jrbarnes9** [comment]
I think `<strong>` or `<b>` would be more suitable for [highlighting](https://ons-design-system.netlify.app/foundations/typography/#highlighting) as we’re using it to distinguish importance and draw attention, and not to change the meaning of a sentence with stressed emphasis.

> The `<strong>` [HTML](https://developer.mozilla.org/en-US/docs/Web/HTML) element indicates that its contents have strong importance, seriousness, or urgency

_https://developer.mozilla.org/en-US/docs/Web/HTML/Element/strong_

> The `<strong>` element is for content that is of greater importance, while the `<b>` element is used to draw attention to text without indicating that it's more important.

_https://developer.mozilla.org/en-US/docs/Web/HTML/Element/strong#b_vs._strong_

**Reply by armstrongb** [reply]
Without changing the font-weight involved in the surrounding sentence?

**Reply by jrbarnes9** [reply]
Depends if we want to override browser defaults if we know they have issues.

**Reply by jrbarnes9** [reply]
Raised #2093

**Comment by armstrongb** [comment]
https://github.com/ONSdigital/design-system/blob/master/src/scss/vars/_typography.scss - font-weight in here is `600`
but `.ons-u-fw-b`, `_header-nav.scss`, `_uac.scss` and `_tabs.scss` set it at `700` and don’t use `$font-weight-bold` unlike in `/_label.scss`

`_autosuggest.scss` uses `font-weight: bold;`

Can we set out what the correct weight truly is for the font and make it consistent?

This could also play a part in what oddities are going on with <em> and .ons-highlight that Noelle is having a hard time with atm.

This [PR - 840](https://github.com/ONSdigital/design-system/pull/840) may also have some part in the cause

**Reply by jrbarnes9** [reply]
Raised #2144

**Comment by jrbarnes9** [comment]
External link component missing visually hidden 'new tab' message for screen readers

See `screenreaderMessage` used in [lists](https://ons-design-system.netlify.app/foundations/typography/#lists)

**Reply by jrbarnes9** [reply]
Raised #2145

**Comment by jrbarnes9** [comment]
DAC: Add space before all vh context set when ‘target: _blank’ set 

And update default text to ‘(opens in a new tab)’ to be consistent with other uses

**Reply by jrbarnes9** [reply]
#2336

**Comment by mr-nick17** [comment]
The utility classes given in the [type scale](https://ons-design-system.netlify.app/foundations/typography/#type-scale) examples are missing the additional 'bem' hyphen `-` e.g `ons-u-fs-s–b` should be `ons-u-fs-s–-b`

**Reply by jrbarnes9** [reply]
Raised #2599
